openweather/src/main/java/pages/PlaceOrderPage.java

```diff
@@ -108,50 +105,57 @@ public PlaceOrderPage(WebDriver driver) {
         this.driver = driver;
     }
 
-    public PlaceOrderPage enterSearch(String searchStr, String searchBy) {
-        waitForPageLoaded();
-        searchTxt.click();
-        searchByPopUp.findElement(By.xpath("//button[text()='" + searchBy + "']")).click();
-        searchTxt.click();
-        searchTxt.sendKeys(searchStr);
-        sleep(2);
-        searchTxt.sendKeys(Keys.DOWN, Keys.RETURN);
-
-        return this;
-    }
-
-    public PlaceOrderPage clickAddLocationButton() {
-        new WebDriverWait(driver, TIMEOUT_IN_SECONDS)
-                .until(ExpectedConditions.visibilityOf(addLocationBtn));
-        addLocationBtn.click();
+    public PlaceOrderPage addLocation(Location location) {
+        if (location != null) {
+            waitForPageLoaded();
+            searchTxt.click();
+            switch (location.searchBy) {
```

Comment:
Don't use `switch case`, it's hard to maintain your code, simply write a private method.